util/saconfig

```diff
@@ -61,74 +62,95 @@ def calibrate(dev, channels):
         logging.info("Calibrating channel %d", channel)
         # set calibration options
         dev.SetProperty_i32(smaract.SA_CTLDLL.SA_CTL_PKEY_CALIBRATION_OPTIONS,
-                        channel, calibration_options)
+                            channel, calibration_options)
 
         dev.Calibrate(channel)
 
 
-def autozero(dev, channels):
+# Reference mode to set
+NORMAL_REF, AUTO_ZERO_REF = 0, 1
+
+
+def set_autozero_reference(dev, channels):
     if not channels:
         return
-
-    # Set auto-zero mode
+    # Set auto-zero referencing mode
     for channel in channels:
         logging.info("Setting auto-zero mode for channel %d", channel)
         dev.SetProperty_i32(smaract.SA_CTLDLL.SA_CTL_PKEY_REFERENCING_OPTIONS,
                             channel, smaract.SA_CTLDLL.SA_CTL_REF_OPT_BIT_AUTO_ZERO)  # auto zero
 
-    # Run the referencing
-    logging.info("Starting reference...")
-    dev.reference().result()
-    time.sleep(0.1)
-    logging.info("Complete")
 
+def set_normal_reference(dev, channels):
+    if not channels:
+        return
     # Set normal referencing mode
-    for channel in  channels:
+    for channel in channels:
         logging.info("Setting the reference mode to normal for channel %d", channel)
         dev.SetProperty_i32(smaract.SA_CTLDLL.SA_CTL_PKEY_REFERENCING_OPTIONS,
                             channel, smaract.SA_CTLDLL.SA_CTL_REF_OPT_BIT_NORMAL)  # normal
 
 
+def reference(dev, channels, mode=NORMAL_REF):
+    # Set referencing mode, currently either normal or autozero
+    if mode == NORMAL_REF:
+        set_normal_reference(dev, channels)
+    elif mode == AUTO_ZERO_REF:
+        set_autozero_reference(dev, channels)
+    # Run the referencing
+    logging.info("Starting reference...")
+    dev.reference().result()
```

Comment:
The driver now needs an "axes" argument. (This got recently broken).
You need to pass a set of the "channels" as strings (because we named the axis as the string version of the channel numbers).